In [36]:
from evaluator import RetrieverEvaluator,compare_models
import json


In [37]:
evaluator = RetrieverEvaluator()


In [38]:
with open('/Users/abdullah/Documents/Embedding-Test/RAG-Embeddings/results_json/cohere_results_comp.json', 'r') as file:
    cohere_data = json.load(file)
cohere_results = evaluator.evaluate_retrieval(cohere_data)


with open('/Users/abdullah/Documents/Embedding-Test/RAG-Embeddings/results_json/voyage_ai_results_comp.json', 'r') as file:
    voyage_data = json.load(file)
voyage_results = evaluator.evaluate_retrieval(voyage_data)

In [39]:
with open('/Users/abdullah/Documents/Embedding-Test/RAG-Embeddings/results_json/gemini_results_comp_528.json', 'r') as file:
    gemini_data = json.load(file)
gemini_results = evaluator.evaluate_retrieval(gemini_data)
with open('/Users/abdullah/Documents/Embedding-Test/RAG-Embeddings/results_json/openai_results_comp.json', 'r') as file:
    openai_data = json.load(file)
openai_results = evaluator.evaluate_retrieval(openai_data)
with open('/Users/abdullah/Documents/Embedding-Test/RAG-Embeddings/results_json/jina_results_comp.json', 'r') as file:
    jina_data = json.load(file)
jina_results = evaluator.evaluate_retrieval(jina_data)

In [40]:
model_results = {
    "Voyage": voyage_results,
    "Cohere": cohere_results,
    "Gemini":gemini_results,
    "OpenAI":openai_results,
    "Jina":jina_results
}


In [41]:
comparison = compare_models(model_results)

In [42]:
print(comparison)

{'hit_ratio_at_3': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'context_length_std': {'best_model': 'Voyage', 'best_score': 11.549350631096104, 'rankings': [('Voyage', 11.549350631096104), ('Cohere', 11.549350631096104), ('Gemini', 11.549350631096104), ('OpenAI', 11.549350631096104), ('Jina', 11.549350631096104)]}, 'hit_ratio_at_1': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'text_diversity': {'best_model': 'Voyage', 'best_score': 0.5630674794316292, 'rankings': [('Voyage', 0.5630674794316292), ('Gemini', 0.40016908049583433), ('OpenAI', 0.3660030484199524), ('Jina', 0.3395213603973389), ('Cohere', 0.3105424255132675)]}, 'hit_ratio_at_5': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'avg_context_l

{'hit_ratio_at_1': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'hit_ratio_at_3': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'avg_context_length': {'best_model': 'Voyage', 'best_score': 113.25, 'rankings': [('Voyage', 113.25), ('Cohere', 113.25), ('Gemini', 113.25), ('OpenAI', 113.25), ('Jina', 113.25)]}, 'hit_ratio_at_5': {'best_model': 'Voyage', 'best_score': 1.0, 'rankings': [('Voyage', 1.0), ('Cohere', 1.0), ('Gemini', 1.0), ('OpenAI', 1.0), ('Jina', 1.0)]}, 'text_diversity': {'best_model': 'Gemini', 'best_score': 0.40554158836603166, 'rankings': [('Gemini', 0.40554158836603166), ('OpenAI', 0.34328250885009765), ('Jina', 0.3395213603973389), ('Cohere', 0.3336178779602051), ('Voyage', 0.3280265361070633)]}, 'context_length_std': {'best_model': 'Voyage', 'best_score': 11.549350631096104, 'rankings': [('Voyage', 11.549350631096104), ('Cohere', 11.549350631096104), ('Gemini', 11.549350631096104), ('OpenAI', 11.549350631096104), ('Jina', 11.549350631096104)]}, 'context_diversity': {'best_model': 'Voyage', 'best_score': -1.0132789611816406e-07, 'rankings': [('Voyage', -1.0132789611816406e-07), ('Cohere', -1.0132789611816406e-07), ('Gemini', -1.0132789611816406e-07), ('OpenAI', -1.0132789611816406e-07), ('Jina', -1.0132789611816406e-07)]}}
